src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,129 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  // #region TextField's
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setimgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  // #endregion
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setimgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setCount(count + 1);
+  };
+
+  const pattern =
+    // eslint-disable-next-line max-len
+    /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const validateForm = () => {
+    const EmptyField =
+      title.trim() === '' ||
+      imgUrl.trim() === '' ||
+      imdbUrl.trim() === '' ||
+      imdbId.trim() === '';
+
+    const isLink =
+      pattern.test(imgUrl.trim()) === true &&
+      pattern.test(imdbUrl.trim()) === true;
+
+    if (EmptyField || !isLink) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const handlerOnSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbId,
+      imdbUrl,
+    });
+
+    resetForm();
+  };
+
+  const getTextFields = () => {
+    if (!validateForm()) {
+      return true;
+    }
+
+    if (title && imgUrl && imdbUrl && imdbId) {
+      return false;
+    }
+
+    return true;
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handlerOnSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={setTitle}
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        name="description"
+        label="Description"
+        value={description}
+        onChange={value => setDescription(value)}
```

Comment:
```suggestion
        onChange={setDescription}
```

The same in other places